products/iam/helpers/ansible/service_account_key_template.erb

```diff
@@ -77,7 +77,7 @@ def create(module):
     auth = GcpSession(module, 'iam')
     json_content = return_if_object(module, auth.post(self_link(module), resource_to_request(module)))
     with open(module.params['path'], 'w') as f:
-        private_key_contents = base64.b64decode(json_content['privateKeyData'])
+        private_key_contents = base64.b64decode(json_content['privateKeyData']).decode('UTF-8')
```

Comment:
```suggestion
        private_key_contents = to_native(base64.b64encode(to_bytes(json_content['privateKeyData'])))
```

Reply:
I'm really confused as to what you're attempting to do here with the changes you suggested, @Akasurde, could you perhaps elaborate?

Looks like you're suggesting we take already base64 encoded data that we needed to decode, and encoding it again...

Google gives us base64, then we decode it (b64decode), it ends up as 'bytes' encoded content, which we then need to get to a string (decode('UTF-8')).

I can see swapping `.decode('UTF-8')` for the `to_native()` function, but the rest I'm confused about .

Reply:
[`to_bytes`](https://github.com/ansible/ansible/blob/7cf0d82108a30e335127e587d900adda9f331557/lib/ansible/module_utils/_text.py#L52) is Ansible-native method to decode string. As code is very explanatory and used widely in [Ansible repo](https://github.com/ansible/ansible/blob/7cf0d82108a30e335127e587d900adda9f331557/lib/ansible/plugins/filter/core.py#L438), I suggest you to use the same.

Reply:
@Akasurde This doesn't explain the switch from b64decode to b64encode at all?

Reply:
to me, it seems like you've completely inverted the situation, thinking we need to get string to bytes, and encode base64 rather than decode it.

Reply:
Ah ! I meant

```python
 private_key_contents = to_native(base64.b64decode(to_bytes(json_content['privateKeyData'])))
```

Sorry for the typo.

Reply:
@Akasurde I changed to using to_native() rather than decode('UTF-8') in my latest commit.

I left out the to_bytes(), as b64decode accepts both strings and bytes in both python2 and 3, so the additional step seems superfluous.